modules/apps/portal/portal-spring-extender-impl/src/main/java/com/liferay/portal/spring/extender/internal/configuration/ServiceConfigurationExtender.java

```diff
@@ -66,10 +71,25 @@ public ServiceConfigurationExtension addingBundle(
 
 		Configuration portletConfiguration = ConfigurationUtil.getConfiguration(
 			classLoader, "portlet");
+
+		if (portletConfiguration != null) {
+			_initModuleResourceActions(classLoader, portletConfiguration);
+
+			BundleContext bundleContext = bundle.getBundleContext();
+
+			_serviceRegistration = bundleContext.registerService(
```

Comment:
@tinatian this does not seem right. So it is registering a Configuration service here, and expecting the removedBundle() to unregister it. However there is a shortcut at https://github.com/shuyangzhou/liferay-portal/pull/10395/commits/c3fefde4e8755f1d881f3686fbb5e61f4968d704#diff-a37965eedced193b103650a7e4bd7c3e2502f774dc2a6467b03d68d1684c516dR93 in case the incoming bundle does not have "service" Configuration, we are not tracking this bundle, therefore no one would unregister this Configuration service, right?

Reply:
@shuyangzhou Yes, you are right, I did not notice this. I will close this pull and resend